safedelete/tests/test_soft_delete.py

```diff
@@ -104,3 +104,28 @@ def test_validate_unique(self):
                 name='test'
             ).validate_unique
         )
+
+    def test_check_unique_fields_exists(self):
+        # No unique fields
+        self.assertEqual(self.instance.has_unique_fields(self.instance), False)
+        # Unique fields
+        unique_model = UniqueSoftDeleteModel.objects.create(
+            name='test-unique'
+        )
+        self.assertEqual(unique_model.has_unique_fields(unique_model), True)
+
+    def test_update_or_create_no_unique_field(self):
+        SoftDeleteModel.objects.update_or_create(id=1)
+        obj, created = SoftDeleteModel.objects.update_or_create(id=1)
+        self.assertEqual(obj.id, 1)
+
+    def test_update_or_create_with_unique_field(self):
+        # Create and soft-delete object
+        obj, created = UniqueSoftDeleteModel.objects.update_or_create(name='unique-test')
+        obj.delete()
+        # Update it and see if it fails
+        obj, created = UniqueSoftDeleteModel.objects.update_or_create(name='unique-test')
+        self.assertEqual(obj.name, 'unique-test')
```

Comment:
Test if created is `True`, @Gagaro what should the behaviour be here? Should a new UUID be created as well and all of the foreign keys deleted?

Reply:
I'm not sure what you mean by that?

Also you could try changing the name here so that we can test that the object is correctly updated.

Reply:
If a model is soft deleted and `update_or_create` is called on it. Should the `created` return value be `True` or `False` cause technically it was not created?

Reply:
I guess `False` because it was not created.

Reply:
Would that not perhaps bring some side effects with it? Some code assuming that if it's created, it should send out an email? I'd play on the safe side and return `True`. This should also be in the docstring.

Reply:
Indeed I'm not sure what the behavior should be here.

Usually, when you have to do some action on an instance you just created, it's because it has just been added to the database. With a soft-deleted object, you may already have what you need.

Having `True` here when the object "exists" might be unexpected. The actual solution would be to have a third value, but that would also complicate things...

Reply:
@Gagaro We could explicitly outline this special case in the docs? And stick to return `False` because technically the object is not created?

Reply:
Yes we should do that. Can you add a line in the doc explaining that?

Reply:
That's all right for me.

Reply:
Nice.